pkg/controller/operators/olm/operator.go

```diff
@@ -449,18 +450,31 @@ func newOperatorWithConfig(ctx context.Context, config *operatorConfig) (*Operat
 		}
 	}
 
-	labelObjects := func(gvr schema.GroupVersionResource, informer cache.SharedIndexInformer, sync queueinformer.LegacySyncHandler) error {
+	complete := map[schema.GroupVersionResource]bool{}
+	completeLock := &sync.RWMutex{}
+
+	labelObjects := func(gvr schema.GroupVersionResource, informer cache.SharedIndexInformer, sync func(done func() bool) queueinformer.LegacySyncHandler) error {
 		if canFilter {
 			return nil
 		}
+		complete[gvr] = false
 		op.k8sLabelQueueSets[gvr] = workqueue.NewRateLimitingQueueWithConfig(workqueue.DefaultControllerRateLimiter(), workqueue.RateLimitingQueueConfig{
 			Name: gvr.String(),
 		})
 		queueInformer, err := queueinformer.NewQueueInformer(
 			ctx,
 			queueinformer.WithLogger(op.logger),
 			queueinformer.WithInformer(informer),
-			queueinformer.WithSyncer(sync.ToSyncer()),
+			queueinformer.WithSyncer(sync(func() bool {
+				completeLock.Lock()
```

Comment:
suggestion: I like the lock() defer unlock() pattern because it's robust against changes b/w the lock() and unlock() calls.

Reply:
I've seen both as a matter of opinion. I prefer the non-defer case in small critical sections with trivial control flow - not sure if true still but also in the past the defer approach was a couple orders of magnitude slower, which is not great when placed into a high-churn callsite.

Reply:
Great call out - I didn't know that. Thank you ^^

Reply:
> not sure if true still but also in the past the defer approach was a couple orders of magnitude slower,

FWIW, this is no longer true. 

From Go 1.14 release notes:
> This release improves the performance of most uses of defer to incur almost zero overhead compared to calling the deferred function directly. As a result, defer can now be used in performance-critical code without overhead concerns.